ols/app/endpoints/streaming_ols.py

```diff
@@ -0,0 +1,208 @@
+
+
+# TODO: more unification of the streaming and non-streaming endpoints,
+# half of the endpoint is practically the same
+@router.post("/streaming_query")
+def conversation_request(llm_request: LLMRequest, auth: Any = Depends(auth_dependency)):
+    """Handle conversation requests for the OLS endpoint.
+
+    Args:
+        llm_request: The request containing a query, conversation ID, and optional attachments.
+        auth: The Authentication handler (FastAPI Depends) that will handle authentication Logic.
+
+    Returns:
+        Response containing the processed information.
+    """
+    timestamps: dict[str, float] = {}
+    timestamps["start"] = time.time()
+
+    # Initialize variables
+    previous_input = []
+
+    user_id = retrieve_user_id(auth)
+    logger.info("User ID %s", user_id)
+    timestamps["retrieve user"] = time.time()
+
+    conversation_id = retrieve_conversation_id(llm_request)
+    timestamps["retrieve conversation"] = time.time()
+
+    # Important note: Redact the query before attempting to do any
+    # logging of the query to avoid leaking PII into logs.
+
+    # Redact the query
+    llm_request = redact_query(conversation_id, llm_request)
+    timestamps["redact query"] = time.time()
+
+    # Log incoming request (after redaction)
+    logger.info("%s Incoming request: %s", conversation_id, llm_request.query)
+
+    previous_input = retrieve_previous_input(user_id, llm_request)
+    timestamps["retrieve previous input"] = time.time()
+
+    # Retrieve attachments from the request
+    attachments = retrieve_attachments(llm_request)
+
+    # Redact all attachments
+    attachments = redact_attachments(conversation_id, attachments)
+
+    # All attachments should be appended to query - but store original
+    # query for later use in transcript storage
+    query_without_attachments = llm_request.query
+    llm_request.query = append_attachments_to_query(llm_request.query, attachments)
+    timestamps["append attachments"] = time.time()
+
+    validate_requested_provider_model(llm_request)
+
+    # Validate the query
+    if not previous_input:
+        valid = validate_question(conversation_id, llm_request)
+    else:
+        logger.debug("follow-up conversation - skipping question validation")
+        valid = True
+
+    timestamps["validate question"] = time.time()
+
+    if not valid:
+        summarizer_response = SummarizerResponse(
+            constants.INVALID_QUERY_RESP,
+            [],
+            False,
+        )
+    else:
+        summarizer_response = generate_response(
+            conversation_id, llm_request, previous_input, streaming=True
+        )
+
+    return StreamingResponse(
+        response_processing_wrapper(
+            summarizer_response,
+            user_id,
+            conversation_id,
+            llm_request,
+            attachments,
+            valid,
+            query_without_attachments,
+            llm_request.stream_text,
```

Comment:
At this point , it is always streaming right ? why do we need this bool ?

Reply:
This affects WHAT is streamed - text or jsons

Reply:
renamed